core/src/parser/mod.rs

```diff
@@ -155,8 +155,44 @@ impl<'ast> Parser<'ast> {
     }
 
     #[inline]
+    fn param_from_expression(&mut self, expression: Expression<'ast>) -> Parameter<'ast> {
```

Comment:
It would probably be slightly faster if the signature was:

```rust
fn param_from_expression(&mut self, expression: ExpressionPtr<'ast>) -> ParameterPtr<'ast>
```

The `FooPtr<'ast>` are just aliases for `Ptr<'ast, Loc<Foo<'ast>>>`, you can copy the original position from the `Loc` struct when transforming them here too :).